Android.mk

```diff
@@ -13,44 +13,54 @@ LOCAL_SRC_FILES := \
     ion_buffer.c \
     common.c
 
+# ---------------- FPC ----------------
 ifeq ($(filter-out loire tone,$(SOMC_PLATFORM)),)
 LOCAL_SRC_FILES += fpc_imp_loire_tone.c
 HAS_FPC := true
-LOCAL_CFLAGS += -DHAS_DYNAMIC_POWER_MANAGEMENT
 endif
 
 ifeq ($(filter-out yoshino,$(SOMC_PLATFORM)),)
 LOCAL_SRC_FILES += fpc_imp_yoshino_nile_tama.c
 HAS_FPC := true
 LOCAL_CFLAGS += \
-    -DUSE_FPC_YOSHINO \
-    -DHAS_DYNAMIC_POWER_MANAGEMENT
+    -DUSE_FPC_YOSHINO
 endif
 
 ifeq ($(filter-out nile,$(SOMC_PLATFORM)),)
+# NOTE: Nile can have either FPC or Egistec
 LOCAL_SRC_FILES += fpc_imp_yoshino_nile_tama.c
 HAS_FPC := true
 LOCAL_CFLAGS += \
     -DUSE_FPC_NILE \
-    -DHAS_DYNAMIC_POWER_MANAGEMENT
+    -DHAS_LEGACY_EGISTEC
 endif
 
 ifeq ($(filter-out tama,$(SOMC_PLATFORM)),)
 LOCAL_SRC_FILES += fpc_imp_yoshino_nile_tama.c
 HAS_FPC := true
 LOCAL_CFLAGS += \
-    -DUSE_FPC_TAMA \
-    -DHAS_DYNAMIC_POWER_MANAGEMENT
+    -DUSE_FPC_TAMA
 endif
 
-ifeq ($(filter-out ganges,$(SOMC_PLATFORM)),)
+# ---------------- Egistec ----------------
+ifeq ($(filter-out nile ganges kumano seine,$(SOMC_PLATFORM)),)
+LOCAL_CFLAGS += -DFINGERPRINT_TYPE_EGISTEC
+
+ifeq ($(filter-out kumano seine,$(SOMC_PLATFORM)),)
+LOCAL_CFLAGS += \
+    -DEGISTEC_SAVE_TEMPLATE_RETURNS_SIZE \
+    -DEGIS_QSEE_APP_NAME=\"egista\" \
+    -DEGIS_QSEE_APP_PATH=\"/odm/firmware\"
+else
 LOCAL_CFLAGS += \
-    -DUSE_FPC_GANGES \
-    -DHAS_DYNAMIC_POWER_MANAGEMENT
+    -DEGIS_QSEE_APP_NAME=\"egisap32\"
 endif
 
-ifeq ($(filter-out kumano,$(SOMC_PLATFORM)),)
-LOCAL_CFLAGS += -DUSE_FPC_KUMANO
+endif
+
+# Define dynamic power management for everything but the following platforms:
+ifneq ($(filter-out kumano seine,$(SOMC_PLATFORM)),)
+LOCAL_CFLAGS += -DHAS_DYNAMIC_POWER_MANAGEMENT
```

Comment:
Maybe group here as well:

HAS_FPC := true

changing the  comment?

Reply:
Then `ganges` needs to be added to the exclusive filter as well, which has power management.

The makefile changes are only for Egistec at this time. I'll likely convert the entire file to BluePrint when R comes around, and further clean it up.